domains/certificates/RegistryConnector/Worker/EventHandlers/CertificateFailedInRegistryEventHandler.cs

```diff
@@ -65,6 +65,13 @@ private async Task Reject(MeteringPointType meteringPointType, Guid certificateI
                 logger.LogWarning("Consumption certificate with certificateId {certificateId} not found.", certificateId);
                 return;
             }
+
+            if (consumptionCertificate.IsRejected)
```

Comment:
This consumption flow and production flow is almost the same. Could it be made generic if we had an abstract Certificate class with a reject method to handle these log things?

Reply:
corrected